aten/src/ATen/native/Attn.cpp

```diff
@@ -0,0 +1,21 @@
+#include <ATen/core/Tensor.h>
+#include <ATen/ATen.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor> attn(const Tensor& query, const Tensor& key, const Tensor &value) {
+    TORCH_CHECK(query.dim() == 2, "expected 2D `query`, got ", query.dim(), "-D tensor")
+    TORCH_CHECK(key.dim() == 2, "expected 2D `key`, got ", key.dim(), "-D tensor")
+    TORCH_CHECK(value.dim() == 2, "expected 2D `value`, got ", value.dim(), "-D tensor")
+    TORCH_CHECK(query.sizes()[1] == key.sizes()[1], "expected size of key to be ", query.sizes()[1], " in last dimension but got: ", key.sizes()[1])
+    
+    // FIXME at::native::matmul or at::matmul?
```

Comment:
When in doubt, always call `at::matmul`.
The difference is that `native::` will call directly into the kernel, bypassing the dispatcher. And so if there is any dispatcher feature needed (like functionalization, negview, etc) it will be bad.